x-pack/filebeat/input/http_endpoint/config.go

```diff
@@ -79,9 +87,29 @@ func (c *config) Validate() error {
 		return errors.New("hmac.type must be sha1 or sha256")
 	}
 
+	if c.CRCProvider != "" {
+		err := validateCRCProvider(c.CRCProvider)
+		if err != nil {
+			return err
+		} else if c.SecretValue == "" {
+			return errors.New("secret.value is required when crc.provider is defined")
+		}
+	}
+
 	return nil
 }
 
+func validateCRCProvider(value string) error {
```

Comment:
This is a binary outcome, so I would return an `ok bool`, rather than an `error`. If if ends up being based on a `map[string]func(string) error` this would become
```
func isValidCRCProvider(name string) (ok bool) {
    _, ok := crcProviders[strings.ToLower(name)]
    return ok
}
```
This does require that the global map be keyed on the lowercase name. That should be documented in the map's docs.